UVicMartletplace/martletplace
# Brent/backend64 setup database tables

# Description

Created the main database tables almost as specified in the data entities section of the SRS 1.0. These are created within the initdb.sql file, where the docker container is initializing the database on each built. I'm curious how to manage persistent storage.

Notes:
- merged reviews and ratings into one ratings tables where reviews (which are a text content) are optional while ratings are not (and default zero)
- any timestamps (not including updated/modified ones) are by default assigned to the  current time (so time of creation)
- deletions of users cascade to SearchHistory, UserPreferences, And RatingTable. (so essentially deleting users won't effect listings and messages)
- dummy data is not added in initdb.sql.
- These tables and attributes are subject to change, but they act as a starting point.

I'm not sure how Resops wants to handle persistant storage, I could've mounted something but I am unsure. It also feels weird to push a big storage mount like that.

Closes #64 

## How to Test
1. close all docker images:
```docker-compose down -v```

2. Run database image:
```docker-compose up -d database```

4. Access postgreSQL server through psql on host:
```psql -p 8310 -U martletplace -h localhost```

5. Enter the password (see the docker-compose.yml for pw, under database)

7. connect to marketplace and view all tables
```
\c martletplace
\dt
```


## Checklist
